### PR TITLE
Kf 4075 feat unit tests for jupyter scipy rock

### PR DESCRIPTION
**Pull request by i-chvets**
Unit tests for Jupyter Scipy ROCK.
These tests verify that all artefacts are in place, that imports can be made in mounted/executed Python script, and that Jupyter Notebook Server is accessible.

Summary of changes:
    - Tests for testing ROCK integrity.
    - Commented out some imports in import test as TO-DOs to be addressed during integation stage. Those might not be required to be tested as imports. Integration should verify this.
    - Changed /home/jovyan to be owned by _daemon_
    - Added notebook server access test.
    - Added unit test to test artifacts, imports, and access.
    - Added tox.ini

Testing:
```
tox -e unit
```

### CODE REVIEW

File: jupyter-tensorflow-cuda-full/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+      && apt-get clean \
+      && rm -rf /var/lib/apt/lists/*
+
+      # tensorflow fix - wrong libcuda lib path (+ reconfigure dynamic linker run-time bindings)
+      ln -s ${CRAFT_PART_INSTALL}/usr/local/cuda/lib64/stubs/libcuda.so ${CRAFT_PART_INSTALL}/usr/local/cuda/lib64/stubs/libcuda.so.1 \
+      && echo "/usr/local/cuda/lib64/stubs" > /etc/ld.so.conf.d/z-cuda-stubs.conf \
+      && ldconfig
+
+      # copy CUDA config
+      mkdri -p ${CRAFT_PART_INSTALL}/etc/ld.so.conf.d/
```
Comment (SamirPS) [inline jupyter-tensorflow-cuda-full/rockcraft.yaml:206]: it's mkdir

File: jupyter-tensorflow-cuda-full/rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+      # Install the jupyter-tensorflow-full requirements
+      cp ${CRAFT_PART_SRC}/components/example-notebook-servers/jupyter-tensorflow-full/requirements.txt requirements.txt
+      "${CONDA_BIN}/pip" install --no-cache-dir -r requirements.txt
+
+      # Create some directories for staging files
+      mkdir -p "${CRAFT_PART_INSTALL}/opt" "${CRAFT_PART_INSTALL}/etc" "${CRAFT_PART_INSTALL}/home"
+
+      # Write some config files for activating conda in user sessions
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${HOME}/.bashrc"
```
Comment (SamirPS) [inline jupyter-tensorflow-cuda-full/rockcraft.yaml:228]: why a space ?

File: jupyter-tensorflow-cuda-full/rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${CRAFT_PART_INSTALL}/etc/profile"
```
Comment (SamirPS) [inline jupyter-tensorflow-cuda-full/rockcraft.yaml:229]: why a space ?

File: jupyter-tensorflow-cuda-full/rockcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+      echo "conda activate base" >> "${HOME}/.bashrc"
+      echo "conda activate base" >> "${CRAFT_PART_INSTALL}/etc/profile"
+
+      # Do some cleanup
+      rm -rf "${HOME}/.wget-hsts" "${HOME}/.cache"
```
Comment (SamirPS) [inline jupyter-tensorflow-cuda-full/rockcraft.yaml:234]: it's hsts or host ?

File: jupyter-tensorflow-cuda-full/rockcraft.yaml (hunk 1 above)
Comment (i-chvets) [reply jupyter-tensorflow-cuda-full/rockcraft.yaml]: Not related file got checked in with #35. Updated.

File: jupyter-tensorflow-cuda-full/rockcraft.yaml (hunk 2 above)
Comment (i-chvets) [reply jupyter-tensorflow-cuda-full/rockcraft.yaml]: Not related file got checked in with #35. Updated.

File: jupyter-tensorflow-cuda-full/rockcraft.yaml (hunk 4 above)
Comment (i-chvets) [reply jupyter-tensorflow-cuda-full/rockcraft.yaml]: Not related file got checked in with #35. Updated.

File: jupyter-tensorflow-cuda-full/rockcraft.yaml (hunk 3 above)
Comment (i-chvets) [reply jupyter-tensorflow-cuda-full/rockcraft.yaml]: Not related file got checked in with #35. Updated.

**Review by SamirPS — APPROVED** [review]

**Comment by SamirPS** [comment]
i have only approve the ``rockcraft.yaml``. I let other for the other files

**Review by ca-scribner — APPROVED** [review]

**Pull request merged**